node/modules/peering/peering.go

```diff
@@ -0,0 +1,45 @@
+package peering
+
+import (
+	"github.com/ipfs/go-ipfs/peering"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type PeeringPeer struct {
+	ID        string   `json:"ID"`
+	Addrs     []string `json:"Addrs"`
+	Connected bool     `json:"Connected,omitempty"`
+}
+
+type EstuaryPeeringService struct {
+	*peering.PeeringService
+}
+
+type EstuaryPeeringNotifee peering.PeeringService
+
+func NewEstuaryPeeringService(host host.Host) *EstuaryPeeringService {
+	estuaryPeeringService := &EstuaryPeeringService{}
+	estuaryPeeringService.PeeringService = peering.NewPeeringService(host)
+	return estuaryPeeringService
+}
+
+func (ps *EstuaryPeeringService) Start() error {
+	return ps.PeeringService.Start()
+}
+
+func (ps *EstuaryPeeringService) Stop() error {
+	return ps.PeeringService.Stop()
+}
+
+func (ps *EstuaryPeeringService) ListPeers() []peer.AddrInfo {
+	return ps.PeeringService.ListPeers()
+}
+
+func (ps *EstuaryPeeringService) AddPeer(info peer.AddrInfo) {
+	ps.PeeringService.AddPeer(info)
+}
+
+func (ps *EstuaryPeeringService) RemovePeer(peerId peer.ID) {
+	ps.PeeringService.RemovePeer(peerId)
+}
```

Comment:
I'm not sure why we need to wrap everything up under the name of `EstuaryPeeringService`. Can't we simply use `ps.PeeringService` altogether?

Reply:
We want to make sure that we are loosely coupled from the implementation so that we can extend the functionality. I wrapped it so that can add more features and `decorate` this later if needed.